### ISSUE TITLE
Go panic in NodeController every 1 second

### ISSUE THREAD

**Issue by dnoland1** [rancher/rancher#25203]
**What kind of request is this (question/bug/enhancement/feature request):**
bug

**Steps to reproduce (least amount of steps as possible):**
unknown

**Result:**
Seeing this about every 1 second in Rancher logs:
```
2020-01-27T04:01:08.803268734Z 2020/01/27 04:01:08 [ERROR] NodeController xxxxxxxxxxxxxxxxx recovered from panic "runtime error: invalid memory address or nil pointer dereference". (err=<nil>) Call stack:
2020-01-27T04:01:08.803306432Z goroutine 443379 [running]:
2020-01-27T04:01:08.803312366Z github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime.RecoverFromPanic(0xc0045a3d30)
2020-01-27T04:01:08.803316668Z  /go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:158 +0xb5
2020-01-27T04:01:08.803320855Z panic(0x3208c60, 0x62c3810)
2020-01-27T04:01:08.803324801Z  /usr/local/go/src/runtime/panic.go:679 +0x1b2
2020-01-27T04:01:08.803350733Z github.com/rancher/rancher/pkg/controllers/user/monitoring.(*clusterMonitoringEnabledHandler).syncWindowsNode(0xc002f03c00, 0xc006b49150, 0xf, 0x0, 0xc000ddd880, 0x7fb72dd1e6d0, 0x0, 0x0)
2020-01-27T04:01:08.803356241Z  /go/src/github.com/rancher/rancher/pkg/controllers/user/monitoring/cluster_monitoring_enabled_handler.go:58 +0xe6
2020-01-27T04:01:08.803360044Z github.com/rancher/rancher/vendor/github.com/rancher/types/apis/core/v1.(*nodeController).AddHandler.func1(0xc006b49150, 0xf, 0x0, 0x0, 0xc0045a3cd8, 0x3, 0x3, 0x0)
2020-01-27T04:01:08.803363969Z  /go/src/github.com/rancher/rancher/vendor/github.com/rancher/types/apis/core/v1/zz_generated_node_controller.go:149 +0xe0
2020-01-27T04:01:08.803387512Z github.com/rancher/rancher/vendor/github.com/rancher/norman/controller.(*genericController).syncHandler(0xc00207a700, 0x2f92320, 0xc008190e00, 0x0, 0x0)
2020-01-27T04:01:08.803390640Z  /go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:367 +0x371
2020-01-27T04:01:08.803393545Z github.com/rancher/rancher/vendor/github.com/rancher/norman/controller.(*genericController).processNextWorkItem(0xc00207a700, 0x0)
2020-01-27T04:01:08.803396147Z  /go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:276 +0xef
2020-01-27T04:01:08.803398815Z github.com/rancher/rancher/vendor/github.com/rancher/norman/controller.(*genericController).runWorker(0xc00207a700)
2020-01-27T04:01:08.803401547Z  /go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:264 +0x2b
2020-01-27T04:01:08.803404977Z github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc00378f670)
2020-01-27T04:01:08.803408494Z  /go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
2020-01-27T04:01:08.803419953Z github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc00378f670, 0x3b9aca00, 0x0, 0x1, 0xc003568ae0)
2020-01-27T04:01:08.803424598Z  /go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
2020-01-27T04:01:08.803428243Z github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc00378f670, 0x3b9aca00, 0xc003568ae0)
2020-01-27T04:01:08.803431733Z  /go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
2020-01-27T04:01:08.803435732Z created by github.com/rancher/rancher/vendor/github.com/rancher/norman/controller.(*genericController).run
2020-01-27T04:01:08.803439922Z  /go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:256 +0xe8
```

**Other details that may be helpful:**
Could be related to https://github.com/rancher/rancher/issues/25164 

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI):
v2.3.4
- Installation option (single install/HA):
HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported):
Custom
- Machine type (cloud/VM/metal) and specifications (CPU/memory):
- Kubernetes version (use `kubectl version`):

```
1.15
```

- Docker version (use `docker version`):

```
?
```

**Comment by deniseschannon** [comment]
This was already fixed in master with https://github.com/rancher/rancher/pull/23985.

**Comment by jiaqiluo** [comment]
The bug is reproduced in v2.3.4 

Steps:
- run rancher v2.3.4 single install
- add an ec2 cluster, at least 2 worker nodes
- enabled the cluster monitoring 
- delete one worker node from the kubectl command-line launched from the UI

Results:
- Rancher logs:
```
2020/02/05 00:04:49 [ERROR] NodeController jiaqi-ec2-234-w3 recovered from panic "runtime error: invalid memory address or nil pointer dereference". (err=<nil>) Call stack:
goroutine 15035 [running]:
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime.RecoverFromPanic(0xc001617d30)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/runtime/runtime.go:158 +0xb5
panic(0x3208c60, 0x62c3810)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/rancher/rancher/pkg/controllers/user/monitoring.(*clusterMonitoringEnabledHandler).syncWindowsNode(0xc0030ed580, 0xc008f8ec60, 0x10, 0x0, 0xc000ee4000, 0x7f448526c008, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/pkg/controllers/user/monitoring/cluster_monitoring_enabled_handler.go:58 +0xe6
github.com/rancher/rancher/vendor/github.com/rancher/types/apis/core/v1.(*nodeController).AddHandler.func1(0xc008f8ec60, 0x10, 0x0, 0x0, 0xc001617cd8, 0x3, 0x3, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/types/apis/core/v1/zz_generated_node_controller.go:149 +0xe0
github.com/rancher/rancher/vendor/github.com/rancher/norman/controller.(*genericController).syncHandler(0xc0049b3700, 0x2f92320, 0xc008fad1d0, 0x0, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:367 +0x371
github.com/rancher/rancher/vendor/github.com/rancher/norman/controller.(*genericController).processNextWorkItem(0xc0049b3700, 0x0)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:276 +0xef
github.com/rancher/rancher/vendor/github.com/rancher/norman/controller.(*genericController).runWorker(0xc0049b3700)
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:264 +0x2b
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc0039665e0)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:152 +0x5e
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc0039665e0, 0x3b9aca00, 0x0, 0x1, 0xc001816e40)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:153 +0xf8
github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait.Until(0xc0039665e0, 0x3b9aca00, 0xc001816e40)
	/go/src/github.com/rancher/rancher/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:88 +0x4d
created by github.com/rancher/rancher/vendor/github.com/rancher/norman/controller.(*genericController).run
	/go/src/github.com/rancher/rancher/vendor/github.com/rancher/norman/controller/generic_controller.go:256 +0xe8
```

**Comment by jiaqiluo** [comment]
The bug fix is validated in v2.3-head 6eecc2c8b and master-head ca01ddff2

the steps are the same.
Results
- no panic in the log 
- a new node is being provisioned and eventually is active
